engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -7635,13 +7635,13 @@ object IR {
 
     /** A representation of an Enso syntax error.
       *
-      * @param at the erroneous AST
+      * @param at the error location
       * @param reason the cause of this error
       * @param passData the pass metadata associated with this node
       * @param diagnostics compiler diagnostics for this node
       */
     sealed case class Syntax(
-      at: AnyRef,
+      at: Option[IdentifiedLocation],
```

Comment:
Yesterday we've been chatting with @hubertp that using `Option` needlessly wastes memory. In JVM the `Option` is really an object and if allocated on heap (or serialized into `.ir`) it occupies 16 bytes and adds one level of pointer indirection hurting L1, L2, L3 caches even more than regular Java pointers.

Can we avoid using (adding new) `Option` into `IR` objects?